packages/@jsii/runtime/lib/sync-stdio.ts

```diff
@@ -61,6 +48,62 @@ export class SyncStdio {
         if (e.code !== 'EAGAIN') {
           throw e;
         }
+        sleep(50 /*ms*/);
+      }
+    }
+  }
+}
+
+/**
+ * A patched up implementation of `fs.readSync` that works around the quirks associated with
+ * synchronous I/O involving `STDIN`, `STDOUT` and `STDERR` on NodeJS, where this is all expected
+ * to be asynchronous and hence has some "interesting" behavior in certain particular cases.
+ *
+ * @param fd        the file descriptor to read from (typically 0 / STDIN)
+ * @param buffer    the buffer in which to place the read data
+ * @param offset    the offset in the buffer at which to place the read data
+ * @param length    the number of bytes to be read
+ * @param position  where to begin reading from the file (defaults to the current read location)
+ *
+ * @returns the amount of bytes read, or `0` if EOF has been reached.
+ */
+function readSync(
+  fd: number,
+  buffer: Buffer,
+  offset: number,
+  length: number,
+  position: number | null = null,
+): number {
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    try {
+      return fs.readSync(fd, buffer, offset, length, position);
+    } catch (error) {
+      switch (error.code) {
+        // HACK: node may set O_NONBLOCK on it's STDIN depending on what kind of input it is made
+        // of (see https://github.com/nodejs/help/issues/2663). When STDIN has O_NONBLOCK, calls may
+        // result in EAGAIN. In such cases, the call should be retried until it succeeds. This kind
+        // of polling will result in horrible CPU thrashing, but there does not seem to be a way to
+        // force a O_SYNC access to STDIN in a reliable way within node.
+        // In order to make this stop we need to either stop depending on synchronous reads, or to
+        // provision our own communication channel that can reliably be synchronous. This work is
+        // "tracked" at https://github.com/aws/aws-cdk/issues/5187
```

Comment:
I see this was copy/pasted; is it still relevant? The referenced issue is closed. Do we have an active/open issue, and/or `jsii` issue to "track" this?